### PR TITLE
Add `GroupNormLayer`

### PR DESCRIPTION
**Pull request by 2wins**
### Checklist
- [x] I've tested that my changes are compatible with the latest version of Tensorflow.
- [x] I've read the [Contribution Guidelines](https://github.com/tensorlayer/tensorlayer/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Description

This is for a new normalization layer, Group Normalization Layer.
Even if there is `tf.contrib.layers.group_norm`, it has an unsolved issue ([link](https://github.com/tensorflow/tensorflow/pull/21390)).

### CODE REVIEW

**Comment by zsdonghao** [comment]
@2wins Thanks. 

Could you also change these two files:

- https://github.com/tensorlayer/tensorlayer/blob/master/docs/modules/layers.rst

- https://github.com/tensorlayer/tensorlayer/blob/master/CHANGELOG.md

**Comment by 2wins** [comment]
> @2wins Thanks.
> 
> Could you also change these two files:
> 
> * https://github.com/tensorlayer/tensorlayer/blob/master/docs/modules/layers.rst
> * https://github.com/tensorlayer/tensorlayer/blob/master/CHANGELOG.md

@zsdonghao OK. I'll reflect it to them.

> It seen this layer only support 2D images?
> 
> ```
> mean, var = tf.nn.moments(x, [1, 2, 4], keep_dims=True)
> ```

@wagamamaz Yes, I only consider 2D input case like `InstanceNormLayer`.

**Comment by wagamamaz** [comment]
@2wins I see, could you raise an exception if the input doesn't fix the shape? 
Also, could we support `channel_first` using `data_format`? Many thanks.

**Comment by zsdonghao** [comment]
@DEKHTIARJonathan I think this one can be merged first, we need to support these kinds of layers, and I have no idea when the 2.0 can be finished..  😭

File: tensorlayer/layers/normalization.py [hunk 1]
```diff
@@ -304,6 +305,126 @@ def __init__(
...
+    """The :class:`GroupNormLayer` layer is for Group Normalization.
+    See `tf.contrib.layers.group_norm <https://www.tensorflow.org/api_docs/python/tf/contrib/layers/group_norm>`__.
+
+    Parameters
+    -----------
+    prev_layer : :class:`Layer`
+        The previous layer.
+    act : activation function
+        The activation function of this layer.
+    others : _
```
Comment (zsdonghao) [inline tensorlayer/layers/normalization.py]: we should put `groups` and `epsilon` here

Comment (2wins) [reply tensorlayer/layers/normalization.py]: It's my mistake. I'll change it. Thanks.

**Comment by 2wins** [comment]
@wagamamaz We need to discuss `data_format`. Many layers we have do not support it normally as can be seen in #640.

**Comment by wagamamaz** [comment]
@2wins the latest version, already support `channel_first`, also the batch norm layer

**Comment by 2wins** [comment]
The commit *Support in GroupNormLayer* https://github.com/tensorlayer/tensorlayer/pull/850/commits/27224ce227b6bbd4d7f3dddb6348ccccda134a8f
should be *Support* `data_format` *in GroupNormLayer*.
